scripts/migration/schema-0.2-GPII-4014/createSimulatedGpiiKeys.js

```diff
@@ -0,0 +1,211 @@
+        var gpiiKeyId = uuid.v4();
+        var prefsSafeId = "prefsSafe-" + gpiiKeyId;
+        var prefs = {
+            "flat": {
+                "contexts": {
+                    "gpii-default": {
+                        "name": "Default preferences",
+                        "preferences": {
+                            "http://registry.gpii.net/common/language": "en-US"
+                        }
+                    }
+                }
+            }
+        };
+
+        var newGpiiKey = {
+            "_id": gpiiKeyId,
+            "type": "gpiiKey",
+            "schemaVersion": "0.1",
+            "prefsSafeId": prefsSafeId,
+            "prefsSetId": "gpii-default",
+            "revoked": false,
+            "revokedReason": null,
+            "timestampCreated": currentTime,
+            "timestampUpdated": null
+        };
+
+        var newPrefsSafe = {
+            "_id": prefsSafeId,
+            "type": "prefsSafe",
+            "schemaVersion": "0.1",
+            "prefsSafeType": "user",
+            "name": gpiiKeyId,
+            "password": null,
+            "email": null,
+            "preferences": prefs,
+            "timestampCreated": currentTime,
+            "timestampUpdated": null
+        };
+        newDocs.push(newGpiiKey);
+        newDocs.push(newPrefsSafe);
+    };
+    return newDocs;
+};
+
+/**
+ * Log how many GPII documents were updated.
+ * @param {String} responseString - Response from the database (ignored)
+ * @param {Object} options - Object containing the set of documents:
+ * @param {Array} options.numOfKeysToCreate - Total number of GPII keys to create.
+ * @param {Array} options.numOfCreatedKeys - Total number of GPII keys that have been created.
+ * @return {Number} - the number of documents created.
+ */
+gpii.migration.GPII4014.logUpdateDB = function (responseString, options) {
+    var togo = fluid.promise();
+    options.numOfCreatedKeys = Number(options.numOfCreatedKeys) + Number(options.numToCreateInThisBatch);
+    if (Number(options.numOfCreatedKeys) < Number(options.numOfKeysToCreate)) {
+        togo.resolve("Created " + options.numOfCreatedKeys + " of requested " + options.numOfKeysToCreate + " GPII keys.");
+    } else {
+        togo.reject({
+            errorCode: "GPII-CREATED-ENOUGH",
+            message: options.numOfCreatedKeys + " have been created."
+        });
+    }
+    return togo;
+};
+
+/**
+ * Configure one batch of creation.
+ * @param {Object} options - Object containing the set of documents:
+ * @param {Array} options.newDocs - The documents to create.
+ * @param {Array} options.numOfKeysToCreate - Total number of GPII keys to create.
+ * @param {Array} options.numOfCreatedKeys - Total number of GPII keys that have been created.
+ * @return {Promise} - The promise that resolves the creation of this batch.
+ */
+gpii.migration.GPII4014.createOneBatch = function (options) {
+    var numOfNeeded = options.numOfKeysToCreate - options.numOfCreatedKeys;
```

Comment:
Why does the previous function have casts and this one not? Probably best to just have the main function driver ensure that all options are numeric when expected